src/types/index.ts

```diff
@@ -140,14 +140,22 @@ export interface Range {
   readonly end: number;
 }
 
+export interface Hunk {
+  readonly oldStart: number;
+  readonly oldEnd: number;
+  readonly newStart: number;
+  readonly newEnd: number;
```

Comment:
I use this object when I am getting the hunks. The hunk output from the `diff` library doesn't produce raw text. It produces a patch text with `"+"` and `"-"` prefixing certain lines and also the comment `"no new line and end of file"` if there is no newline and the end of the file. Therefore I'd need to alter the patch text at this step. So either way I'd need to do a raw text lookup to avoid finicky string manipulation. Therefore I have 2 choices: get and store raw text before I filter the hunks or after. If I do it after the filtration, I save on memory.